saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/RelyingPartyRegistration.java

```diff
@@ -438,10 +440,12 @@ private static Saml2X509Credential fromDeprecated(
 
 		private final Saml2MessageBinding singleSignOnServiceBinding;
 
+		private String signAlgorithmUri;
```

Comment:
So that this aligns better with `<IDPSSODescriptor>`, will you please change this to be a `Collection<String>`? The reason is that an asserting party can specify more than one algorithm. Also, will you please call it `signingAlgorithms` since that's closer to what it's called [in the `<Extensions>` element.](https://www.oasis-open.org/committees/download.php/51890/SAML%20MD%20simplified%20overview.pdf).

Reply:
I do not get this, how can an asserting party have more than one algorithm ?

Reply:
An asserting party can support more than one signing method, listing them in order of preference in their `<IDPSSODescriptor>`. The reliant party should use the highest one on the list that they can.

For example, an asserting party can say "relying parties can either sign with RSA-SHA256 or ECDSA-SHA256". So, if a relying party can use RSA-SHA256, it should, since that was listed first.

OpenSAML ships with support for this negotiation step which you can invoke by setting all the configured algorithms in `BasicSignatureSigningConfiguration#setSignatureAlgorithms`.

Reply:
got it, thanks